### PR TITLE
logs: distinguish WARN and INFO for `peer disconnected`

### LINKED ISSUE #3182: peer disconnected not a warn
steps to reproduce: start neogo from the command line and stop it with ctrl-c:
```
^C2023-10-31T20:11:54.815+0300	INFO	shutting down server	{"peers": 11}
2023-10-31T20:11:54.815+0300	WARN	peer disconnected	{"addr": "170.64.152.72:51333", "error": "server shutdown", "peerCount": 10}
2023-10-31T20:11:54.815+0300	WARN	peer disconnected	{"addr": "65.108.90.74:51333", "error": "server shutdown", "peerCount": 9}
2023-10-31T20:11:54.815+0300	WARN	peer disconnected	{"addr": "159.203.28.78:51333", "error": "server shutdown", "peerCount": 8}
2023-10-31T20:11:54.815+0300	WARN	peer disconnected	{"addr": "49.12.226.190:50333", "error": "server shutdown", "peerCount": 7}
2023-10-31T20:11:54.815+0300	WARN	peer disconnected	{"addr": "5.78.81.79:51333", "error": "server shutdown", "peerCount": 6}
2023-10-31T20:11:54.815+0300	WARN	peer disconnected	{"addr": "49.12.226.190:51333", "error": "server shutdown", "peerCount": 5}
2023-10-31T20:11:54.815+0300	WARN	peer disconnected	{"addr": "170.64.152.72:50333", "error": "server shutdown", "peerCount": 4}
2023-10-31T20:11:54.815+0300	WARN	peer disconnected	{"addr": "116.203.249.217:50333", "error": "server shutdown", "peerCount": 3}
2023-10-31T20:11:54.815+0300	WARN	peer disconnected	{"addr": "167.71.212.120:50333", "error": "server shutdown", "peerCount": 2}
2023-10-31T20:11:54.815+0300	WARN	peer disconnected	{"addr": "159.203.28.78:50333", "error": "server shutdown", "peerCount": 1}
2023-10-31T20:11:54.815+0300	WARN	peer disconnected	{"addr": "65.108.90.74:50333", "error": "server shutdown", "peerCount": 0}
2023-10-31T20:11:55.201+0300	INFO	persisted to disk	{"blocks": 1367, "keys": 22958, "headerHeight": 353191, "blockHeight": 353191, "took": "385.464896ms"}
```

`peer disconnected` is not a warning!

### PR DESCRIPTION
**Pull request by AliceInHunterland**
Peer disconnections are not warnings.

Close #3182

some errors that can be in the `peer disconnected` message:
```
err = msg.Decode(r)
errors.New("invalid payload size")
br.ReadBytes(m.compressedPayload)

fmt.Errorf("bind address for connection '%s' is not registered", localAddr.String())
errors.New("invalid handshake: already sent Version")
_, err = p.conn.Write(b) 

fmt.Errorf("handling %s message: %w", msg.Command.String(), err)
err = p.conn.SetWriteDeadline(time.Now().Add(writeTimeout))

s.requestBlocks(bq, p)
b, err := msg.Bytes()

	errGone           = errors.New("the peer is gone already")
	errStateMismatch  = errors.New("tried to send protocol message before handshake completed")
	errPingPong       = errors.New("ping/pong timeout")
	errUnexpectedPong = errors.New("pong message wasn't expected")  
```

For me, such messages look more like warnings:
```
read: can't assign requested address 
unexpected empty payload: CMDVersion
invalid inventory type
```

### CODE REVIEW

File: pkg/network/server.go [hunk 1]
```diff
@@ -512,10 +512,17 @@ func (s *Server) run() {
 			if s.peers[drop.peer] {
 				delete(s.peers, drop.peer)
 				s.lock.Unlock()
-				s.log.Warn("peer disconnected",
-					zap.Stringer("addr", drop.peer.RemoteAddr()),
-					zap.Error(drop.reason),
-					zap.Int("peerCount", s.PeerCount()))
+				if errors.Is(drop.reason, errInvalidInvType) {
```
Comment (AnnaShaleva) [inline pkg/network/server.go:515]: I'd like to add one more error to this list: the error when a node receives a message with unknown type. In general, it's not a warning itself, but almost all the nodes in the Neo network have "proper" implementation and this case may be alarming.

Comment (AnnaShaleva) [reply pkg/network/server.go:515]: I'll attach the exactly error once my node is sync up with testnet.

**Review by AnnaShaleva — COMMENTED** [review]
Also, I think that `s.requestBlocks(bq, p)`deserves a warning, it's quite critical for the node functioning.

File: pkg/network/server.go (hunk 1 above)
Comment (AnnaShaleva) [reply pkg/network/server.go]: > I'll attach the exactly error once my node is sync up with testnet.

OK, not this time, the peer emitting wrong messages is gone and I haven't saved the exact message to find it in our codebase.

File: pkg/network/server.go [hunk 2]
```diff
@@ -793,7 +802,7 @@ func (s *Server) requestBlocksOrHeaders(p Peer) error {
 	}
 	err := s.requestBlocks(bq, p)
 	if err != nil {
-		return err
+		return fmt.Errorf("%w: failed to request blocks: %w", errInvalidBlock, err)
```
Comment (AnnaShaleva) [inline pkg/network/server.go:805]: It's not "invalid block", it's literally "blocks request failed", so create a separate error with "block request failed" text and use `return fmt.Errorf("%w: %w", errBlocksRequestFailed, err)` here.

File: pkg/network/server.go [hunk 3]
```diff
@@ -514,7 +514,7 @@ func (s *Server) run() {
 			if s.peers[drop.peer] {
 				delete(s.peers, drop.peer)
 				s.lock.Unlock()
-				if errors.Is(drop.reason, errInvalidInvType) || errors.Is(drop.reason, errStateMismatch) || strings.Contains(drop.reason.Error(), errInvalidBlock.Error()) {
+				if strings.Contains(drop.reason.Error(), errInvalidInvType.Error()) || errors.Is(drop.reason, errStateMismatch) || strings.Contains(drop.reason.Error(), errInvalidBlock.Error()) {
```
Comment (AnnaShaleva) [inline pkg/network/server.go:517]: But why?

Comment (AliceInHunterland) [reply pkg/network/server.go:517]: > errInvalidInvType should be extended: we need to format this error to include the inventory time itself. Implement it in a separate commit.

so I changed `errInvalidInvType` to `fmt.Errorf("%w: %s", errInvalidInvType, inv.Type.String())` and that why we need compare strings

Comment (AnnaShaleva) [reply pkg/network/server.go]: `errors.Is` should work with it.

**Review by AnnaShaleva — APPROVED** [review]

**Pull request merged**